kairon/shared/utils.py

```diff
@@ -1609,44 +1626,44 @@ def save_and_publish_auditlog(document, name, **kwargs):
         except AttributeError:
             action = kwargs.get("action")
 
-        audit = Utility.get_auditlog_id_and_mapping(document)
+        audit = [{"key": "Bot_id", "value": Utility.get_auditlog_id_and_mapping(document)}]
 
-        audit_log = AuditLogData(audit=audit,
+        audit_log = AuditLogData(metadata=audit,
                                  user=document.user,
                                  action=action,
                                  entity=name,
                                  data=document.to_mongo().to_dict())
-        Utility.publish_auditlog(auditlog=audit_log, event_url=kwargs.get("event_url"))
         audit_log.save()
+        Utility.publish_auditlog_after_saving(audit_log, event_url=kwargs.get("event_url"))
+
 
     @staticmethod
     def get_auditlog_id_and_mapping(document):
         try:
             auditlog_id = document.bot.__str__()
-            mapping = "Bot_id"
         except AttributeError:
             auditlog_id = document.id.__str__()
-            mapping = f"{document._class_name.__str__()}_id"
 
-        return {mapping: auditlog_id}
+        return auditlog_id
 
     @staticmethod
     def publish_auditlog(auditlog, **kwargs):
         from .data.data_objects import EventConfig
         from mongoengine.errors import DoesNotExist
 
         try:
-            if auditlog.audit.get("Bot_id") is None:
-                logger.debug("Only bot level event config is supported as of")
-                return
-            event_config = EventConfig.objects(bot=auditlog.audit.get("Bot_id")).get()
-
-            headers = json.loads(Utility.decrypt_message(event_config.headers))
-            ws_url = event_config.ws_url
-            method = event_config.method
-            if ws_url:
-                Utility.execute_http_request(request_method=method, http_url=ws_url,
-                                             request_body=auditlog.to_mongo().to_dict(), headers=headers, timeout=5)
+            for data in auditlog.metadata:
```

Comment:
this retrieved bot id to get the event config from db. this event config contains a url to which logs are pushed. 
please revisit this logic